libs/filameshio/src/MeshReader.cpp

```diff
@@ -217,14 +219,14 @@ MeshReader::Mesh MeshReader::loadMeshFromBuffer(filament::Engine* engine,
     Part* parts = (Part*) p;
     p += header->parts * sizeof(Part);
 
-    uint32_t materialCount = (uint32_t) *p;
+    const uint32_t materialCount = *(const uint32_t *)p;
     p += sizeof(uint32_t);
 
-    std::vector<std::string> partsMaterial(materialCount);
+    std::string materials[materialCount];
```

Comment:
You can't do that. Variable size static arrays is a compiler extension, it's not standard.

Reply:
The difference really won't matter. Keep as-is.

Reply:
But the prerequisites for building Filament say clang 14.0 (or more recent) according to the BUILDING.md file.
This compiler allows such allocation.

Reply:
It doesn't matter, we should stick to the standard. std::vector works just fine here, there's no reason to change this.

Reply:
OK. It's a choice.

Reply:
It matters if you don't want to waste memory.